scripts/pre-commit

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+#
+#  Copyright (C) 2022
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+'''Git pre-commit hook to update the copyright year in Sherpa code files
+'''
+from __future__ import print_function
+import datetime
+import fileinput
+import re
+import subprocess
+import sys
+
+year = str(datetime.datetime.now().year)
+copyr = re.compile(r"(?P<comment>(#|//))\s*Copyright \(C\) (?P<years>[0-9,\s]+)(?P<institution>[\w\s]*)")
+# Only apply this to C or Python files since sherpa does not use
+# copyright statements in others files, also, do not change in extern
+# directory.
+ftype = re.compile(r"^((?!extern).)*(hh|cc|py)$")
```

Comment:
For completeness, do you want to include C/C++? The comment says C but the regexp only checks for C++. The code base includes various C files outside of the 'extern directory' (for example in sherpa/utils).

Reply:
Makes sense. I forgot that when I switched form checking all (which can mess up binary files) to just a selected set of files, so added C files.